modules/templates/helper.go

```diff
@@ -313,20 +303,53 @@ func NewFuncMap() []template.FuncMap {
 			htmlCode += "</span>"
 			return template.HTML(htmlCode)
 		},
-		"MermaidMaxSourceCharacters": func() int {
-			return setting.MermaidMaxSourceCharacters
+
+		// -----------------------------------------------------------------
+		// misc
+		"DiffLineTypeToStr":        DiffLineTypeToStr,
+		"ShortSha":                 base.ShortSha,
+		"ActionContent2Commits":    ActionContent2Commits,
+		"IsMultilineCommitMessage": IsMultilineCommitMessage,
+		"CommentMustAsDiff":        gitdiff.CommentMustAsDiff,
+		"MirrorRemoteAddress":      mirrorRemoteAddress,
+
+		"ParseDeadline": func(deadline string) []string {
+			return strings.Split(deadline, "|")
 		},
-		"Join":        strings.Join,
-		"QueryEscape": url.QueryEscape,
-		"DotEscape":   DotEscape,
-		"Iterate": func(arg interface{}) (items []int64) {
-			count, _ := util.ToInt64(arg)
-			for i := int64(0); i < count; i++ {
-				items = append(items, i)
+		"FilenameIsImage": func(filename string) bool {
+			mimeType := mime.TypeByExtension(filepath.Ext(filename))
+			return strings.HasPrefix(mimeType, "image/")
+		},
+		"TabSizeClass": func(ec interface{}, filename string) string {
+			var (
+				value *editorconfig.Editorconfig
+				ok    bool
+			)
+			if ec != nil {
+				if value, ok = ec.(*editorconfig.Editorconfig); !ok || value == nil {
+					return "tab-size-8"
+				}
+				def, err := value.GetDefinitionForFilename(filename)
+				if err != nil {
+					log.Error("tab size class: getting definition for filename: %v", err)
+					return "tab-size-8"
+				}
+				if def.TabWidth > 0 {
+					return fmt.Sprintf("tab-size-%d", def.TabWidth)
+				}
 			}
-			return items
+			return "tab-size-8"
```

Comment:
On an unrelated note: I've never understood why our default tab size is `8`, why not something more normal such as `4`?

Reply:
I agree tab size 4 default is better. But it's probably not the only change necessary and should be in another PR.